exporter/googlecloudexporter/config.go

```diff
@@ -32,39 +28,36 @@ import (
 const (
 	defaultMetricPrefix = "workload.googleapis.com"
 	defaultUserAgent    = "observIQ-otel-collector"
-	defaultLocation     = "global"
-	genericNodeResource = "generic_node"
 )
 
 // Config is the config the google cloud exporter
 type Config struct {
 	config.ExporterSettings `mapstructure:",squash"`
-	Credentials             string                                       `mapstructure:"credentials"`
-	CredentialsFile         string                                       `mapstructure:"credentials_file"`
-	Location                string                                       `mapstructure:"location"`
-	Namespace               string                                       `mapstructure:"namespace"`
-	GCPConfig               *gcp.LegacyConfig                            `mapstructure:",squash"`
-	BatchConfig             *batchprocessor.Config                       `mapstructure:"batch"`
-	NormalizeConfig         *normalizesumsprocessor.Config               `mapstructure:"normalize"`
-	DetectorConfig          *resourcedetectionprocessor.Config           `mapstructure:"detector"`
-	AttributerConfig        *resourceprocessor.Config                    `mapstructure:"attributer"`
-	TransposerConfig        *resourceattributetransposerprocessor.Config `mapstructure:"transposer"`
```

Comment:
This is a breaking change and will break existing users who are currently using these removed fields on update (from my understanding); What's our plan on migrating them?

Reply:
It's actually not as much of a breaking change as you'd expect. We removed quite a few embedded processors, but no customers or any of our sample configs were actually configuring those values before. So we don't foresee any problems with those removals.

The two biggest breaking changes will be removing location and namespace as config options. We had to do that because the newest exporter no longer supports custom resource mappings. Instead, it automatically converts resources and if nothing matches, it falls back to generic node. Based on the logic in that conversion, the location and namespace can be defined by the `cloud.region` and `service.namespace` attributes respectively. So going forward, users will need to use a processor to configure this, instead of the exporter.

The field team has already been notified about this and reviewed the differences between the two exporters yesterday afternoon. DevOps also verified that the newest exporter worked without having to make changes to their configs.

Reply:
I think what concerns me is that we have sample configs like:
https://github.com/observIQ/observiq-otel-collector/blob/9c5c21a40635d6a4cb5a3184b0c333b26d891c21/config/google_cloud_exporter/redis/config.yaml

Which are broken (haven't tested it, but it should fail to unmarshal since resource_mappings is no longer a valid key).

We might consider fixing up these configs "follow-up" work, but I feel like we should have these sample configs in a working state before release. (Unless I'm completely wrong and these configs work fine)